srml/support/src/storage/generator.rs

```diff
@@ -101,20 +103,19 @@ pub trait Storage {
 #[cfg(feature = "std")]
 impl<S: sr_primitives::BuildStorage> Storage for (crate::rstd::cell::RefCell<&mut sr_primitives::StorageOverlay>, PhantomData<S>) {
 	fn exists(&self, key: &[u8]) -> bool {
-		self.0.borrow().contains_key(S::hash(key).as_ref())
+		UnhashedStorage::exists(self, &S::hash(key))
 	}
 
 	fn get<T: codec::Decode>(&self, key: &[u8]) -> Option<T> {
-		self.0.borrow().get(S::hash(key).as_ref())
-			.map(|x| codec::Decode::decode(&mut x.as_slice()).expect("Unable to decode expected type."))
+		UnhashedStorage::get(self, &S::hash(key))
 	}
 
 	fn put<T: codec::Encode>(&self, key: &[u8], val: &T) {
-		self.0.borrow_mut().insert(S::hash(key).to_vec(), codec::Encode::encode(val));
+		UnhashedStorage::put(self, &S::hash(key), val)
```

Comment:
a hash clone could be avoided here, probably not worth changing the code.

Reply:
which clone ? you mean it clones `S::hash(key)` ? we use a reference no ?

Reply:
in fact it is the same clone as the to_vec() from previous code (would need a into_vec to possibly remove a clone which is not currently possible with Hash::Out). Please forget this comment, sorry for the noise.

Reply:
no problem :-)